pkg/reconciler/reconciler_test.go

```diff
@@ -216,6 +214,46 @@ func TestReconcile(t *testing.T) {
 				g.Expect(mock.finalizeInvoked).To(BeFalse())
 			},
 		},
+		{
+			name: "requeues on conflict",
+			objects: []client.Object{
+				&v1alpha1.Istio{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       key.Name,
+						Finalizers: []string{testFinalizer},
+					},
+				},
+			},
+			setup: func(g *WithT, mock *mockReconciler) {
+				mock.reconcileError = apierrors.NewConflict(schema.GroupResource{}, "foo", fmt.Errorf("simulated conflict"))
+			},
+			assert: func(g *WithT, cl client.Client, result ctrl.Result, err error, mock *mockReconciler) {
+				g.Expect(result).To(Equal(reconcile.Result{RequeueAfter: 2 * time.Second}))
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(mock.reconcileInvoked).To(BeTrue())
+				g.Expect(mock.finalizeInvoked).To(BeFalse())
+			},
+		},
+		{
+			name: "requeues when gc admission plugin does not yet know about our resources",
+			objects: []client.Object{
+				&v1alpha1.Istio{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       key.Name,
+						Finalizers: []string{testFinalizer},
+					},
+				},
+			},
+			setup: func(g *WithT, mock *mockReconciler) {
+				mock.reconcileError = apierrors.NewForbidden(schema.GroupResource{}, "foo", fmt.Errorf("cannot set blockOwnerDeletion in this case because cannot find RESTMapping for APIVersion ..."))
```

Comment:
@dgn Do you remember if this was the error that we were seeing?

Reply:
yes, pretty sure that was it